R/paginate.R

```diff
@@ -18,38 +18,69 @@
 #'
 #'   * a new [request] to request the next page or
 #'   * `NULL` if there is no next page.
+#' @param page_size A whole number that specifies the page size i.e. the number
+#'   of elements per page.
+#' @param set_page_size A function applies the page size to the request. It must
+#'   have the arguments `req` and `page_size`.
 #' @param n_pages A function that extracts the total number of pages from
-#'   the [response].
+#'   the [response]. If the `page_size` argument is provided, the `page_size`
```

Comment:
I'm not sure this is easy enough to understand.

Reply:
Yeah, as is, it doesn't feel like a big win over just repeating the `page_size` in multiple places.